### ISSUE TITLE
Support backward compatibility to create application roles via scim2/Roles endpoint

### ISSUE THREAD

**Issue by isuruhettiarachchi** [wso2/product-is#20741]
**Describe the issue:**
In IS 6.1.0, application role creation via the scim2/Roles endpoint was removed, along with the application role validation when creating and accessing service providers. However, application role validation can be enabled to maintain backward compatibility with the following config.

```
[application_mgt]
enable_role_validation=true
```

With the same config, it is required to allow the users to create application roles via scim2/Roles endpoint.

**How to reproduce:**

1. Create an application role via scim2/Roles endpoint
```
curl -L 'https://localhost:9443/scim2/Roles' \
-H 'Authorization: Basic YWRtaW46YWRtaW4=' \
-H 'Content-Type: application/json' \
-d '{
    "displayName": "Application/testrolescb900",
    "permissions": [
        "/permission/admin/login",
        "/permission/admin/manage/identity/applicationmgt",
        "/permission/admin/manage/identity/claimmgt"
    ],
    "schemas": []
}'
```
2. It will response the following
```
{
    "schemas": [
        "urn:ietf:params:scim:api:messages:2.0:Error"
    ],
    "scimType": "Invalid character: / contains in the role name: Application/testrolescb900.",
    "detail": "Request is unparsable, syntactically incorrect, or violates schema.",
    "status": "400"
}
```

**Expected behavior:**
Users should be able to create application roles when `enable_role_validation` config is enabled.

**Environment information:**
 - Product Version: IS 6.1.0

**Comment by nilasini** [comment]
Seems we have decided not to fix this improvement to the public branch not to encourage the creation of application roles. In that case did we add this information to the documentation? Don't we need to deprecate the configuration support? @SujanSanjula96 @sadilchamishka

**Comment by SujanSanjula96** [comment]
@nilasini Yes. we decided to not to give this functionality in master. `enable_role_validation` this configuration is used to give the backward compatibility for the previous validation support as well. We need to decide whether we are deprecating that option for migrating apps.

cc: @DMHP
